### PR TITLE
test(x/ecocredit): basket integration test

### PR DESCRIPTION
**Pull request by technicallyty**
## Description

Closes: #XXXX

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by technicallyty** [comment]
theres some mismatch between impl's here so i think i need #772 to reconcile this PR

**Review by aaronc — APPROVED** [review]

File: x/ecocredit/server/server.go [hunk 1]
```diff
@@ -83,12 +85,7 @@ func RegisterServices(
 	bankKeeper ecocredit.BankKeeper,
 	distKeeper ecocredit.DistributionKeeper,
 ) {
-	impl := newServer(configurator.ModuleKey(), paramSpace, accountKeeper, bankKeeper, configurator.Marshaler())
-	db, err := ormutil.NewStoreKeyDB(ModuleSchema, configurator.ModuleKey(), ormdb.ModuleDBOptions{})
-	if err != nil {
-		panic(err)
-	}
-	impl.basketKeeper = basket.NewKeeper(db, impl, bankKeeper, distKeeper, impl.storeKey)
```
Comment (robert-zaremba) [inline x/ecocredit/server/server.go:91]: :+1:  for moving to the constructor

File: x/ecocredit/server/testsuite/suite.go [hunk 2]
```diff
@@ -976,3 +1190,34 @@ func (s *IntegrationTestSuite) TestScenario() {
...
+		Admin:          admin.String(),
+		Issuers:        []string{admin.String()},
+		Metadata:       nil,
+		CreditTypeName: creditType,
+	})
+	require.NoError(err)
+	classId := cRes.ClassId
+	start, err := time.Parse("2006-04-02", startStr)
+	require.NoError(err)
+	end, err := time.Parse("2006-04-02", endStr)
```
Comment (robert-zaremba) [inline x/ecocredit/server/testsuite/suite.go:1209]: we should have a function to do parsing instead of repeating the format string.

**Review by robert-zaremba — APPROVED** [review]
utACK

**Review by ryanchristo — APPROVED** [review]
Nice work! Looks good to me.

**Pull request merged**